src/ert/_c_wrappers/job_queue/forward_model.py

```diff
@@ -56,7 +58,38 @@ def get_job_data(
     ) -> Dict[str, Any]:
         def substitute(job, string):
             if string is not None:
-                string = job.private_args.substitute(string)
+                copy_private_args = copy.deepcopy(job.private_args)
+                # The user might have already added <ITER> and <IENS> as
+                # expicit arguments, in which case we shouldn't touch them. The
+                # weird "passing through" of arguments needs to be preserved
+                # but deprecated.
+                if (
+                    "<ITER>" in copy_private_args
+                    and copy_private_args["<ITER>"] == "<ITER>"
+                ):
+                    warn(
+                        f"Found <IENS>=<IENS> in {job.name} forward model."
+                        " This argument can safely be removed and setting it in this "
+                        " way is deprecated. Using for instance <ITER>=1 is still "
+                        " possible."
+                    )
+                    copy_private_args.addItem("<ITER>", str(itr))
+                if (
+                    "<IENS>" in copy_private_args
+                    and copy_private_args["<IENS>"] == "<IENS>"
+                ):
+                    warn(
+                        f"Found <IENS>=<IENS> in {job.name} forward model."
+                        " This argument can safely be removed and setting it in this"
+                        " way is deprecated. Using for instance <IENS>=1 is still "
+                        " possible."
+                    )
+                    copy_private_args.addItem("<IENS>", str(iens))
+                if "<ITER>" not in copy_private_args:
+                    copy_private_args.addItem("<ITER>", str(itr))
+                if "<IENS>" not in copy_private_args:
+                    copy_private_args.addItem("<IENS>", str(iens))
```

Comment:
This is not really needed

Reply:
Not really, but there are conditions where copy_private_args.substitute may not terminate if these are not added. Those uses are deprecated of course.